inference-engine/src/gna_plugin/transformations/decompose_2d_conv.cpp

```diff
@@ -0,0 +1,672 @@
+    //size_t pool_stride_height;
+};
+
+struct ConvParams {
+    size_t input_height;
+    size_t input_width;
+    size_t input_channel_count;
+    size_t filter_height;
+    size_t filter_width;
+    size_t filter_count;
+    size_t filter_channel_count;
+    size_t filter_dilation_height;
+    size_t filter_dilation_width;
+    size_t filter_stride_height;
+    size_t filter_stride_width;
+    size_t output_height;
+    size_t output_width;
+    size_t output_channel_count;
+    ngraph::element::Type element_type;
+};
+
+static bool VerifyAndGetConvParams(std::shared_ptr<ngraph::opset7::Convolution> conv, ConvParams& conv_params) {
+    const auto& input = conv->input_value(0);
+    const auto& filters = conv->input_value(1);
+
+    // We support only 2D conv batch 1
+    if (conv->get_dilations().size() != 2 ||
+        conv->get_strides().size() != 2 ||
+        input.get_shape()[0] != 1) {
+        return false;
+    }
+
+    size_t filter_height = filters.get_shape()[2];
+    size_t filter_width = filters.get_shape()[3];
+
+    if (filter_width > GNALimitations::copyMaxGrouping || filter_height > GNALimitations::copyMaxGrouping) {
+        return false;
+    }
+
+    conv_params.output_height = conv->get_output_shape(0)[2];
+    conv_params.output_width = conv->get_output_shape(0)[3];
+    conv_params.input_channel_count = conv->input_value(0).get_shape()[1];
+    conv_params.input_height = conv->input_value(0).get_shape()[2];
+    conv_params.input_width = conv->input_value(0).get_shape()[3];
+    conv_params.filter_count = conv->input_value(1).get_shape()[0];
+    conv_params.filter_channel_count = conv->input_value(1).get_shape()[1];
+    conv_params.filter_height = conv->input_value(1).get_shape()[2];
+    conv_params.filter_width = conv->input_value(1).get_shape()[3];
+    conv_params.filter_dilation_height = conv->get_dilations()[0];
+    conv_params.filter_dilation_width = conv->get_dilations()[1];
+    conv_params.filter_stride_height = conv->get_strides()[0];
+    conv_params.filter_stride_width = conv->get_strides()[1];
+    conv_params.output_channel_count = conv_params.filter_count;
+    conv_params.element_type = conv->get_element_type();
+
+    IE_ASSERT(conv_params.output_channel_count == conv->get_output_shape(0)[1]);
+
+    return true;
+}
+
+static bool TransposeOrderMatches(std::shared_ptr<ngraph::opset7::Transpose> transpose, std::vector<int64_t> order) {
+    if (!transpose)
+        return false;
+    const ngraph::Output<ngraph::Node>& transpose_order = transpose->input_value(1);
+    auto transpose_order_dim = transpose_order.get_shape().size();
+
+    if (transpose_order_dim != 1 || transpose_order.get_shape()[0] != order.size())
+        return false;
+
+    auto const_with_order_values = std::dynamic_pointer_cast<ngraph::opset7::Constant>(transpose_order.get_node_shared_ptr());
+    if (!const_with_order_values)
+        return false;
+
+    const int64_t* data = const_with_order_values->get_data_ptr<int64_t>();
```

Comment:
I have already commented on this on other PR. Could you please refactor code to avoid duplication of code and duplication of comments for this code?

Reply:
I tried to make fixes in both transforms, but must have missed this one. Anyway, I'll create a helper file to avoid code duplication.

Reply:
@sirzabek otherwise convolution is not supposed to be valid